AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/office/ManageOfficeViewModel.kt

```
@@ -0,0 +1,93 @@
+import com.android.agrihealth.data.model.office.Office
+import com.android.agrihealth.data.model.office.OfficeRepository
+import com.android.agrihealth.data.model.user.Vet
+import com.android.agrihealth.ui.user.UserViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+data class ManageOfficeUiState(
+    val office: Office? = null,
+    val editableName: String = "",
+    val editableDescription: String = "",
+    val editableAddress: String = "",
+    val isLoading: Boolean = false,
+    val error: String? = null
+)
+
+class ManageOfficeViewModel(
+    private val userViewModel: UserViewModel,
+    private val officeRepository: OfficeRepository
+) : ViewModel() {
+
+  private val _uiState = MutableStateFlow(ManageOfficeUiState())
+  val uiState: StateFlow<ManageOfficeUiState> = _uiState
+
+  init {
+    loadOffice()
+  }
+
+  fun loadOffice() {
+    viewModelScope.launch {
+      _uiState.value = _uiState.value.copy(isLoading = true)
+
+      val currentUser = userViewModel.user.value
+      if (currentUser is Vet && currentUser.officeId != null) {
+        val office = officeRepository.getOffice(currentUser.officeId).getOrNull()
+        _uiState.value =
+            _uiState.value.copy(
+                office = office,
+                editableName = office?.name ?: "",
+                editableDescription = office?.description ?: "",
+                editableAddress = office?.address?.toString() ?: "")
+      }
+
+      _uiState.value = _uiState.value.copy(isLoading = false)
+    }
+  }
+
+  fun onNameChange(value: String) {
+    _uiState.value = _uiState.value.copy(editableName = value)
+  }
+
+  fun onDescriptionChange(value: String) {
+    _uiState.value = _uiState.value.copy(editableDescription = value)
+  }
+
+  fun onAddressChange(value: String) {
+    _uiState.value = _uiState.value.copy(editableAddress = value)
+  }
+
+  fun leaveOffice(onSuccess: () -> Unit) =
```

make sure to handle the case the userViewModel updates successfully but not the office repository, especially since the userViewModel fails silently for now.